### PR TITLE
feat(#356): bnpl schema, models and feature flag

### PR DESCRIPTION
**Pull request by robwilde**
Foundation sub-task of epic #355. Everything else in the epic depends on these tables existing.

**Spec:** `docs/superpowers/specs/2026-07-25-bnpl-email-planned-payments-design.md`
**Plan:** `docs/superpowers/plans/2026-07-25-bnpl-schema-models-flag.md`

## What

- `bnpl_orders` — one row per parsed BNPL schedule email, current state plus full email provenance
- `bnpl_order_events` — append-only audit trail, shaped after the existing `pipeline_audit_entries`
- `BnplProvider`, `BnplOrderStatus`, `BnplOrderEventType` enums
- `budget.bnpl_email_import` flag, default **false**

No production code reads these yet. #357–#362 do.

## Why two unique constraints

`unique(user_id, gmail_message_id)` makes a re-scan a no-op.

`unique(user_id, provider, order_ref)` is the one that matters for correctness: a provider resending a confirmation gives the same order a **new message id**, and without this that copy would create a second `PlannedTransaction` and double the forecast. This constraint came out of spec review, not the original design.

## Review notes

- **`frequency` is nullable.** An unevenly spaced schedule must persist for review rather than be forced into a cadence that misrepresents it. #357 leaves it null, #359 routes it to `pending_review` with a `review_note`.
- **`isSettled()` is exclusive of today.** A schedule whose final instalment falls today is still *live* and must still produce a planned transaction. Three tests pin that boundary directly, because the obvious `isPast()` assertion disagrees with the method at exactly that point.
- **`BnplOrderEventType` is an enum** although the issue text says only provider and status are. Four sub-tasks write audit rows and a mistyped event value is silent and permanent.
- **`recordEvent()` and `isSettled()` ship with no caller.** Deliberate: one audit write path and one settled predicate, rather than four divergent copies across #359–#362.
- **The factory scopes its account to the order's own user.** `Account::factory()->for($user)` does *not* do this — Laravel resolves the `user_id` attribute and the `for()` relationship independently, so the default fixture produced `order.user_id=14` with `account.user_id=15`. Verified empirically, and there is now a regression test. `BankImportFactory` and `PlannedTransactionFactory` carry the same latent bug; left alone as out of scope.
- **No migration touches `planned_transactions` or `transactions`** — explicit acceptance criterion of the issue and the epic.
- Declined a suggested `User::bnplOrders()` inverse: `BankImport`, the closest analogue, has none either, and no caller exists yet.

## Verification

- Full suite: **2018 passed, 5034 assertions**
- New coverage: 22 `BnplOrderTest` + 7 `BnplOrderEventTest` + 15 enum unit tests
- Pint PASS and PHPStan `[OK] No errors` across all 15 changed PHP files
- `migrate:rollback --step=2` drops both tables with no FK error, `migrate` re-applies
- Live MySQL confirms both unique indexes
- Rector skipped — the project has no `rector.php` and the command only offers to generate one

Refs #356

### CODE REVIEW

**Comment by robwilde** [comment]
**Re: Copilot's factory comment** (https://github.com/robwilde/can-eye-budget-v2/pull/364#discussion_r3650277832)

Your initial premises need refinement, but the edge case is real and now fixed.

1. **make() purity**: Already false upstream—`'user_id' => User::factory()` resolves via `create()` in both contexts. Moving to `afterCreating` buys zero purity.

2. **user_id timing**: For normal usage, Laravel's `expandAttributes()` resolves in passes; `user_id` precedes `account_id`, so the closure receives a persisted integer.

**The real issue**: `withoutParents()` / `withoutRelationships()` set `user_id => null`, and the unguarded closure would attempt `Account::factory()->create(['user_id' => null])`.

**Fix**: Inline closure with guard clause now prevents account creation on null `user_id`. Keeps `definition()` efficient and `make()` behavior unchanged. All 26 BNPL tests pass.

File: database/factories/BnplOrderFactory.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            'parsed_payload' => null,
+            'reviewed_at' => null,
+        ];
+    }
+
+    public function autoApproved(): self
+    {
+        return $this->state(fn (array $attributes) => [
+            'status' => BnplOrderStatus::AutoApproved,
+            'category_id' => Category::factory(),
```
[the remark this answers is not in the document]
Comment (robwilde) [reply database/factories/BnplOrderFactory.php]: Both premises here were tested empirically against this repo and don't hold, so the closure stays as-is (no source change):

- `BnplOrder::factory()->make()` already writes `users=1 accounts=1 bnpl=0` — `make()` purity is unreachable while `'user_id' => User::factory()` sits in the definition, which is the convention every factory in `database/factories/` follows. Moving only the account write to `afterCreating()` wouldn't make `make()` pure; it would leave `account_id` null on `make()` and add an extra `UPDATE` on `create()`.
- The pattern this suggestion drifts toward already exists in the repo and is broken: `BankImportFactory::definition()` resolves `$user = User::factory()` once but is referenced twice, so `import.user_id` and `account.user_id` end up on two different users (measured on both default `create()` and `->for($u)`). Not copying that.
- The null-`user_id` guard added in `304053f` (this factory, lines 40-42) is the real edge-case fix, and it now has a dedicated regression test: `withoutParents makes an unsaved order and creates no user or account` in `tests/Feature/Models/BnplOrderTest.php`, alongside a new coverage test for the `budget.bnpl_email_import` flag default in `tests/Feature/Config/BudgetFeatureFlagsTest.php`. Both were verified to fail when their guarded behavior is reverted, and the full BNPL module (43 tests) plus the whole suite, Pint, and PHPStan are green on top of these changes.

Resolving; happy to reopen if there's a case the guard doesn't cover.

File: app/Models/BnplOrderEvent.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * @property BnplOrderEventType $event
+ * @property array<string, mixed>|null $payload
+ * @property string|null $actor
+ * @property CarbonImmutable $created_at
+ * @property CarbonImmutable $updated_at
+ */
+final class BnplOrderEvent extends Model
+{
+    /** @use HasFactory<BnplOrderEventFactory> */
+    use HasFactory;
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Models/BnplOrderEvent.php:26]: Checked this against the codebase's own precedent this PR is modelled on. `PipelineAuditEntry` (`app/Models/PipelineAuditEntry.php`) — the model `bnpl_order_events` is explicitly shaped after — has no immutability guard either, and neither do `BasiqRefreshLog`, `PipelineRun`, or `BankImport`. 'Append-only' is a documented convention enforced by discipline (nothing in the codebase calls `update()`/`delete()` on these logs), not a technical guard, anywhere in this repo.

Adding a first-of-its-kind immutability enforcement to `BnplOrderEvent` alone would introduce a novel pattern with no precedent, for a table that no code writes to or reads from yet (#357–#362, which build the writers, aren't merged). That's out of scope for issue #356 (schema/models/flag) and not one of its acceptance criteria.

Not implementing here. Tracking as a candidate follow-up once a real writer exists in #357+ — happy to revisit if the team wants immutability enforced repo-wide as its own change.

Separately: fixed the suggestion on `tests/Feature/Config/BudgetFeatureFlagsTest.php` — added the missing `@noinspection StaticClosureCanBeUsedInspection` header to match every other Pest file (`63b7536`). The three stale-count nitpicks against `docs/superpowers/plans/2026-07-25-bnpl-schema-models-flag.md` are left alone — that plan doc is a historical record of what ran during development, not living documentation.
